src/libsync/propagatedownloadzsync.cpp

```diff
@@ -124,62 +117,56 @@ bool GETFileZsyncJob::finished()
     return true; // discard
 }
 
-void GETFileZsyncJob::seedFinished(void* zs) {
-    _zs = (struct zsync_state *)zs;
+void GETFileZsyncJob::seedFinished(void* zs)
+{
+    zsync_unique_ptr<struct zsync_state> _zs((struct zsync_state*)zs, [](struct zsync_state* zs) {
```

Comment:
You're shadowing the member `_zs` here, probably unintended!

Reply:
You're not looking at latest push?